homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,243 @@
+"""Platform for switch integration."""
+from __future__ import annotations
+
+from boschshcpy import (
+    SHCCamera360,
+    SHCCameraEyes,
+    SHCLightSwitch,
+    SHCSession,
+    SHCSmartPlug,
+    SHCSmartPlugCompact,
+)
+from boschshcpy.device import SHCDevice
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+
+from .const import DATA_SESSION, DOMAIN
+from .entity import SHCEntity
+
+SWITCH_TYPES: dict[str, SwitchEntityDescription] = {
```

Comment:
This is a good start - but I think you can move `is_on` / `turn_on` / `turn_off` / `toggle` to the base class by adding two properties (`on_key` and `on_value`) to the EntityDescription :

```python
    @property
    def is_on(self):
        """Return the state of the switch."""
        return getattr(self._device, self.entity_description.on_key) == self.entity_description.on_value

    def turn_on(self, **kwargs):
        """Turn the switch on."""
        setattr(self._device, self.entity_description.on_key, True)

    def turn_off(self, **kwargs):
        """Turn the switch off."""
        setattr(self._device, self.entity_description.on_key, False)

    def toggle(self, **kwargs):
        """Toggle the switch."""
        setattr(self._device, self.entity_description.on_key, not self.is_on)
```

You can take a look at Renault binary sensor as an EntityDescription example: https://github.com/home-assistant/core/blob/38723b277e05e498e4defb816cb47e4e5d63bfd1/homeassistant/components/renault/binary_sensor.py#L24-L29

After you have added `on_key` and `on_value`, you can add `has_consumption` to enable `today_energy_kwh` and `current_power_w`